src/primitives/line/mod.rs

```diff
@@ -209,6 +209,21 @@ impl Line {
             return Intersection::Colinear;
         }
 
+        let outer_side = if denominator > 0 {
+            LineSide::Right
+        } else {
+            LineSide::Left
+        };
+
+        // Special case: If the two lines are almost parallel, return the average point between
+        // them.
+        if denominator.pow(2) < self.delta().dot_product(other.delta()) {
+            return Intersection::Point {
+                point: (self.end + other.start) / 2,
```

Comment:
Isn't this dependent on the order of `star` and `end`? If, for example, `self.end` is close to the intersection and `other.start` is further away than `other.end` the average won't be in the expected place.

Reply:
That's true, however for the nearly-parallel case*, each point will only be 1 or 2 px away from the other so it doesn't matter. I originally just went for using `point: self.end`.

* Which I'm hoping is the only time the `if` is entered into.

Reply:
I've added a PR for a debug tool to the `debug-tools` repo. If you move `l2_end` vertically, with all other points at their default value, you'll notice that the intersection point doesn't move, because it averages the wrong points.

Reply:
And this case isn't correct either:
![grafik](https://user-images.githubusercontent.com/226123/105636150-2debc280-5e67-11eb-903c-ea2ad4db317f.png)